run_build_impl.sh

```diff
@@ -286,6 +291,41 @@ cd $WORKSPACE
 # Prepare cppcheck ignore list. We want to skip dependencies.
 CPPCHECK_PARAMS="src --xml --enable=missingInclude,performance,style,portability,information -j8 -ibuild -i$DEPS"
 
+function kill_roscore_on_exit {
+  if $START_ROSCORE && $ROS_PID -gt 0 ; then
```

Comment:
I think you are missing a `[[  ..  ]] ` around `$ROS_PID -gt 0` . And I'd suggest a `""` around `$ROS_PID` to be robust against an empty value for `ROS_PID`